CoordgenMinimizer.cpp

```diff
@@ -1177,7 +1177,7 @@ bool CoordgenMinimizer::flipFragments(sketcherMinimizerMolecule* molecule,
     vector<sketcherMinimizerFragment*> fragments = molecule->getFragments();
     reverse(fragments.begin(), fragments.end());
     for (auto fragment : fragments) {
-        if (!fragment->fixed) {
+        if (!fragment->fixed && !fragment->constrained) {
```

Comment:
this makes sense to me, but the way I envision it is to assign a penalty in CoordgenFlipFragmentDOF::getCurrentPenalty()
in the case of constrained fragments. That way the flip is possible but only performed if there's no better alternative

Reply:
@ZontaNicola Many thanks for looking into this. Makes sense to me; I'd make the penalty rather high though. such that the flipping becomes a last-resort option. It looks like most users would rather live with a small clash but keep the layout aligned to the template as much as possible.

Reply:
I agree, I think it would be preferable to lose the alignement to avoid crossing bonds for instance, but not to avoid small clashes